### PR TITLE
Add uniqueidentifier Column Support

### PR DESCRIPTION
**Pull request by wsugarman**
## Description
- Add `UniqueIdentifierColumn` and `NullableUniqueIdentifierColumn` to better support code-gen for `uniqueidentifier`

## Related issues
[AB#83611](https://microsofthealth.visualstudio.com/Health/_workitems/edit/83611)

## Testing
Pass unit tests

## Semver Change ([docs](https://github.com/microsoft/healthcare-shared-components/blob/master/docs/Versioning.md))
Feature

### CODE REVIEW

File: src/Microsoft.Health.SqlServer/Features/Schema/Model/Column.cs [hunk 1]
```diff
@@ -550,6 +569,33 @@ public override void Set(SqlDataRecord record, int ordinal, byte? value)
         }
     }
 
+    public class NullableUniqueIdentifierColumn : Column<Guid?>
```
Comment (brendankowitz) [inline src/Microsoft.Health.SqlServer/Features/Schema/Model/Column.cs:572]: Quick question: does this also mean only 1 row can have the 'null' identifier. Does this make sense?

Comment (wsugarman) [reply src/Microsoft.Health.SqlServer/Features/Schema/Model/Column.cs:572]: I'm sorry, could you rephrase your question? Like the other nullable column types in this file, I'm pretty sure they refer to a representation of some table column type that has been annotated as nullable.

For example, given the SQL table definition:
```SQL
CREATE TABLE Foo
(
    -- ...
    Bar INT NULL,
    -- ...
)
```

In C# we'll code-gen the corresponding field for its table type:
```csharp
internal readonly NullableIntColumn Bar = new NullableIntColumn("Bar");
```

But multiple columns can be annotated as "nullable" for a given row, and multiple rows can contain `NULL` values in these "nullable" columns.

Comment (brendankowitz) [reply src/Microsoft.Health.SqlServer/Features/Schema/Model/Column.cs:572]: Initially I was thinking there was also a uniqueness constraint here, where in re-reading, this change for `UniqueIdentifier` is just the datatype. Sorry for the confusion :)

**Review by brendankowitz — APPROVED** [review]
:shipit:

File: src/Microsoft.Health.SqlServer/Features/Schema/Model/Column.cs (hunk 1 above)
Comment (wsugarman) [reply src/Microsoft.Health.SqlServer/Features/Schema/Model/Column.cs:572]: Ah no problem! Yeah -- it seems that we're missing support for the type `uniqueidentifier`, so this PR is adding it

**Comment by wsugarman** [comment]
I have written the `[Nullable]UniqueIdentifierColumn` types to operate on the C# type `Guid` instead of `SqlGuid`. It seems we're already using `Guid` when generating stored procedures with parameters of type `uniqueidentifier`, so I think this change is consistent. The only difference is in how these type perform their comparison operations, detailed [here](https://docs.microsoft.com/en-us/sql/connect/ado-net/sql/compare-guid-uniqueidentifier-values?view=sql-server-ver15).
